targets/TARGET_ONSEMI/TARGET_NCS36510/gpio_irq_api.c

```diff
@@ -195,32 +200,40 @@ void gpio_irq_set(gpio_irq_t *obj, gpio_irq_event event, uint32_t enable)
 {
 
     /* Enable the GPIO clock */
-    CLOCK_ENABLE(CLOCK_GPIO);
+    if(!CLOCK_IS_ENABLED(CLOCK_GPIO)) {
+        CLOCK_ENABLE(CLOCK_GPIO);
+    }
 
     switch(event) {
         case IRQ_RISE:
-            obj->GPIOMEMBASE->IRQ_EDGE = (obj->pinMask);
-            obj->GPIOMEMBASE->IRQ_LEVEL = (IO_ALL ^ (obj->pinMask));
-            /* Enable is an integer; hence checking for 1 or 0*/
+            obj->GPIOMEMBASE->IRQ_EDGE = obj->pinMask;
+
+            /* Enable rising edge */
+            obj->GPIOMEMBASE->IRQ_POLARITY_SET = obj->pinMask;
+
+            /* Enable the IRQ based on enable parameter */
             if (enable == 1) {
-                /* Enable rising edge */
-                obj->GPIOMEMBASE->IRQ_POLARITY_SET = (obj->pinMask);
+
```

Comment:
My comment still applies - you're repeating the same enable code in each case of the switch - the enable code does not depend on the mode, and it doesn't need to be in the switch.